modules/porous_flow/src/utils/PorousFlowVanGenuchten.C

```diff
@@ -609,4 +609,268 @@ d2saturationHys(Real pc,
   }
   return d2s;
 }
+
+Real
+cubic(Real x, Real x0, Real y0, Real y0p, Real x1, Real y1, Real y1p)
+{
```

Comment:
Should you check that `x1 > x0`, maybe an assert?

Reply:
Certainly going to fail if `x1=x0`

Reply:
Included an assert, and made a new `PorousFlowCubic` namespace as suggested